### DISCUSSION TITLE
Trying to build 9.0 for PicoW

### DISCUSSION THREAD

**Discussion by Sudrien** [astlouys/Pico-Green-Clock#18] [Q&A]
I'm on Manjaro, So I installed Pico SDK from AUR

> export PICO_SDK_PATH=/usr/share/pico-sdk

and the standard non-w version builds & runs as expected. Do the uncommenting as instructed, i.e., 

> #define PICO_W

And get the build error

> /home/sudrien/Pico-Green-Clock/picow_ntp_client.h:11:10: fatal error: pico/cyw43_arch.h: No such file or directory
>   11 | #include "pico/cyw43_arch.h"

Which is present, alone, in /usr/share/pico-sdk/src/rp2_common/pico_cyw43_arch/include/pico/cyw43_arch.h
....

I presume CMakeLists.txt.PicoW is also supposed to replace CMakeLists.txt , and the README has not been updated to reflect that yet?

**Comment by Sudrien** [comment]
Yeah, once I knew what I was looking for in the Pdf, it was easy to find.

**Reply by astlouys** [reply]
Hi Sudrien,

Please. don't let me believe that I wrote this 100+pages User Guide for nothing !!! (smile) (In the U.S - where you live - they use to say "RTFM"... but we don't know each other well enough. I won't say it. Don`t get me wrong, I'm just kidding - another smile).

I hope you'll enjoy this firmware. Let me know...

André

**Reply by Sudrien** [reply]
Forgive me, I believe I had a few too many soldier fumes yesterday - I definitely reviewed the 8.0 User guide. 

A few notes;

- I was not able to set SSID and password by the three button method - or at least, there was no indication it saved to flash, including when I went to set it manually a second and third time. astlouys/Pico-Flash-Utility - as recommended - did save me. Adding SSID somewhere in the date scroll - or if the default is set - would be appreciated.

- "NTP Errors: 1" does show up in the date scroll. Looking at the code - this only resets when the firmware is flashed?

- Appendix M, The United States (or at least a majority of it) is a '10' at the moment. 

- I believe I will try installing a passive buzzer shortly.

**Reply by astlouys** [reply]
Hi Sudrien,

1) I'm not sure what went wrong when you tried to "set-and-save" the credentials with the three clock buttons. While reading your text, it seems that you went through the whole process without problem (?)  Here are a few hints... First of all, it's a good idea to set the clock "time and date" to a wrong value. This way, you will see if the time and date are properly re-adjusted a few seconds after power-up cycle is over. Make sure that DST country and timezone are properly set since they have an impact on NTP re-sync. 

Also, once you have set both SSID and Password, if you enter in the "set-and-save credentials" once again (by holding middle button while powering-up), you will see the current SSID and Password scroll on clock display (the first time, you saw only the "footprints: ,;,;,;,;,; and ,:,:,:,:,:,:,:,:". I thought about putting the credentials in the date scroll but changed my mind since it means that all your friends (and / or other people at home) will know your credentials if they take a look at your clock. But as mentioned, you know now that there is a way to check them. (There is also an easier way with a remote control, but I know that not everyone has added an infrared sensor. That's why I added it to the "set-and-save" function.

The Flash is written ("updated") just at the beginning of the next date scroll. So if you entered the credentials and then triggered a power-up cycle (Off / On) right after, before at least one date scroll happens, the credentials were not written to flash (so, NTP won't work).

2) "NTP Errors: 1" means that there was one error while trying to re-sync with NTP. Since NTP re-sync is something that happens "behind the scene", I thought it would be a good idea to show somewhere that something goes wrong. Otherwise, you may have changed the credentials on your router, if you don't remember to change it on the Green Clock, you will never know that re-sync is not happening any more. So, one error is not a problem, but if you see this number increasing as days pass, there could be a problem. This number is reset at each power-up (so if it is "1" every time you start the clock, it probably means that NTP doesn't work).

3) I don't understand how I have missed U.S.A. in the list... I added it in the User Guide that will be released with Version 10.0

4) A passive buzzer is fun, although it would be fun to have someone with "musical ears" to provide more jingles. It is only a "one-frequency" type of music, but it's fun anyway. You probably already read the section in the manual about it...
